### PR TITLE
Add change network button if user uses different network

### PR DESCRIPTION
**Pull request by T-Damer**
- We can probably upgrade the `networkPick()` to include all the needed info to use it in the future

### CODE REVIEW

File: src/stores/WalletStore.ts [hunk 1]
```diff
@@ -95,6 +102,47 @@ class WalletStore extends PersistableStore {
...
+        : `https://${network}.etherscan.io/`
+
+    try {
+      await provider.send('wallet_switchEthereumChain', [{ chainId }])
+      this.needNetworkChange = false
+    } catch (error) {
+      const code = serializeError(error).code
+      if (code !== 4902) return
+
+      await provider.send('wallet_addEthereumChain', [
```
Comment (T-Damer) [inline src/stores/WalletStore.ts:130]: Works if the user doesn't have our network (bnb wallet, for example)

File: .env.sample [hunk 2]
```diff
@@ -1,6 +1,7 @@
 VITE_ENCRYPT_KEY=000000000000000000000000000000000000000000
 VITE_APP_NAME=Seal-Cred
-VITE_ETH_NETWORK=rinkeby
+VITE_ETH_NETWORK=goerli
+VITE_CHAIN_ID=5
```
Comment (smyzin) [inline .env.sample:4]: Can we escape setting `chain_id`?

File: src/stores/WalletStore.ts [hunk 3]
```diff
@@ -95,6 +102,48 @@ class WalletStore extends PersistableStore {
...
+        {
+          chainId,
+          rpcUrls: [`https://${network}.infura.io/v3/`],
+          chainName: `${networkName} Test Network`,
+          nativeCurrency: {
+            name: currency,
+            symbol: currency,
+          },
+          blockExplorerUrls: [blockExplorerUrl],
+        },
```
Comment (smyzin) [inline src/stores/WalletStore.ts:141]: Maybe we can pass it to a separate file? What do you think?

File: src/stores/WalletStore.ts [hunk 4]
```diff
@@ -95,6 +102,48 @@ class WalletStore extends PersistableStore {
...
+      network === 'mainnet'
+        ? 'https://etherscan.io/'
+        : `https://${network}.etherscan.io/`
+
+    try {
+      await provider.send('wallet_switchEthereumChain', [{ chainId }])
+      this.needNetworkChange = false
+    } catch (error) {
+      const code = serializeError(error).code
+      if (code !== 4902) return
```
Comment (smyzin) [inline src/stores/WalletStore.ts:129]: Don't we need to set `needNetworkChange` to `true` here?

**Review by smyzin — APPROVED** [review]
Left several comments, otherwise looks good

File: .env.sample (hunk 2 above)
Comment (T-Damer) [reply .env.sample:4]: Oh, forgot to remove it

File: src/stores/WalletStore.ts (hunk 4 above)
Comment (T-Damer) [reply src/stores/WalletStore.ts:129]: it's already set in `#L110`

**Pull request merged**